compiler/modules/front/token.bal

```diff
@@ -282,6 +304,80 @@ class Tokenizer {
         return self.err("invalid token");
     }
 
+    private function nextDecimalFPNumber(string preDot) returns Token|err:Syntax {
+        string digits = preDot + ".";
+        while true {
+            Char? ch = self.getc();
+            if ch is () {
+                break;
+            }
+            else if DIGIT.includes(ch) {
+                digits += ch;
+            }
+            else if ch == "e" || ch == "E" {
+                return self.nextDecimalExpedFPNumber(digits, ch);
+            }
+            else if ch == "f" || ch == "F" {
+                return [DECIMAL_FP_NUMBER, checkpanic float:fromString(digits)];
+            }
+            else {
+                self.ungetc(ch);
+                break;
+            }
+        }
+        return [DECIMAL_FP_NUMBER, digits];
+    }
+
+    private function nextDecimalExpedFPNumber(string preExp, Char e) returns Token|err:Syntax {
```

Comment:
No idea what `exped` means

Reply:
I mend to say "Exp-ed" meaning "Exponentiated", I'll go with "nextDecimalExpFPNumber" instead.